packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -109,30 +84,16 @@ export class App extends Root {
   }
 
   /**
-   * Returns metadata for all constructs in the stack.
+   * Synthesize the app manifest (the root file which the toolkit reads)
    */
-  public collectMetadata(stack: Stack) {
-    const output: { [id: string]: MetadataEntry[] } = { };
-
-    visit(stack);
-
-    // add app-level metadata under "."
-    if (this.node.metadata.length > 0) {
-      output[PATH_SEP] = this.node.metadata;
-    }
-
-    return output;
-
-    function visit(node: IConstruct) {
-      if (node.node.metadata.length > 0) {
-        // Make the path absolute
-        output[PATH_SEP + node.node.path] = node.node.metadata.map(md => node.node.resolve(md) as MetadataEntry);
-      }
+  protected synthesize(session: ISynthesisSession) {
+    const manifest: cxapi.SynthesizeResponse = {
+      version: cxapi.PROTO_RESPONSE_VERSION,
+      stacks: this.synthesizeStacks(Object.keys(this.stacks)),
```

Comment:
If you `synthesizeStacks()` can go into `ISynthesisSession` as `getStacks()` and then we can just use that here.

Reply:
As mentioned above `SynthesisSession` doesn't know about stacks, only about files.
This is also here since the toolkit reads it from here and not from the files emitted during synthesis. I'll iterate on this further in a subsequent PR.